fdbserver/MoveKeys.actor.cpp

```diff
@@ -373,6 +374,186 @@ ACTOR Future<Optional<UID>> checkReadWrite(Future<ErrorOr<GetShardStateReply>> f
+				actors.push_back(store(UIDtoTagMap, tr.getRange(serverTagKeys, CLIENT_KNOBS->TOO_MANY)));
+				wait(waitForAll(actors));
+				ASSERT(!UIDtoTagMap.more && UIDtoTagMap.size() < CLIENT_KNOBS->TOO_MANY);
+				completeRangeKS = Standalone(KeyRangeRef(rangeToRead.begin, readResultKS.back().key));
+				TraceEvent(SevVerbose, "ValidateRangeAssignmentKeyServersReadDone")
+				    .detail("Range", completeRangeKS)
+				    .detail("ResultSize", readResultKS.size());
+
+				// Read serverKeys
+				actors.clear();
+				state int i = 0;
+				for (; i < readResultKS.size() - 1; ++i) {
+					std::vector<UID> src;
+					std::vector<UID> dest;
+					UID srcID;
+					UID destID;
+					decodeKeyServersValue(UIDtoTagMap, readResultKS[i].value, src, dest, srcID, destID);
+					std::vector<UID> servers(src.size() + dest.size());
+					std::merge(src.begin(), src.end(), dest.begin(), dest.end(), servers.begin());
+					skToReadRange = Standalone(KeyRangeRef(readResultKS[i].key, readResultKS[i + 1].key));
+					for (auto& ssid : servers) {
+						actors.push_back(store(readResultSKs[ssid],
+						                       krmGetRanges(&tr,
+						                                    serverKeysPrefixFor(ssid),
+						                                    skToReadRange,
+						                                    CLIENT_KNOBS->KRM_GET_RANGE_LIMIT,
+						                                    CLIENT_KNOBS->KRM_GET_RANGE_LIMIT_BYTES)));
+					}
+					wait(waitForAll(actors));
+
+					// For each ss which has completeRangeKS indicated by KeyServers
+					// Check: the ss must contain completeRangeKS indicated by ServerKeys
+					for (auto& [ssid, resSK] : readResultSKs) {
+						ASSERT(readResultSKs[ssid].size() == 2 && readResultSKs[ssid].back().key == skToReadRange.end);
+						UID shardId;
+						bool assigned, emptyRange;
+						EnablePhysicalShardMove enablePSM = EnablePhysicalShardMove::False;
+						decodeServerKeysValue(readResultSKs[ssid][0].value, assigned, emptyRange, enablePSM, shardId);
+						if (!assigned) {
+							TraceEvent(SevError, "ValidateLocationMetadataError")
+							    .detail("Context", context)
+							    .detail("AuditRange", skToReadRange)
+							    .detail("ErrorMessage", "KeyServers has range but ServerKeys does not have")
+							    .detail("CurrentEmptyRange", emptyRange)
+							    .detail("CurrentAssignment", assigned)
+							    .detail("EnablePSM", enablePSM)
+							    .detail("ServerID", ssid)
+							    .detail("ShardID", shardId);
+							anyError = true;
+						}
+					}
+				}
+				// Return result
+				if (anyError) {
+					TraceEvent(SevError, "ValidateLocationMetadataError")
+					    .detail("Context", context)
+					    .detail("AuditRange", rangeToCompare);
+					return false;
+				} else {
+					if (completeRangeKS.end < rangeToCompare.end) {
+						rangeToReadBegin = completeRangeKS.end;
+						iterationCount++;
+						if (now() - beginTime > 1) {
```

Comment:
What is this timeout of `1` second for?

Will this `timeout` be caught by the last `catch(` and treated as a `audit_storage_failed`?